src/app/leaves/components/LeavesRequestList.tsx

```diff
@@ -15,95 +21,58 @@ import {
-    startDate: "2024-11-01",
-    endDate: "2024-11-03",
-    status: LeavesRequestStatus.Approved,
-    reason: "Flu",
-    requestedBy: "John Doe",
-  },
-  {
-    id: 2,
-    leaveType: "Vacation",
-    startDate: "2024-12-10",
-    endDate: "2024-12-20",
-    status: LeavesRequestStatus.Pending,
-    reason: "Family vacation",
-    requestedBy: "Jane Smith",
-  },
-  {
-    id: 3,
-    leaveType: "Personal Leave",
-    startDate: "2024-11-15",
-    endDate: "2024-11-16",
-    status: LeavesRequestStatus.Approved,
-    reason: "Personal matter",
-    requestedBy: "Emily Johnson",
-  },
-  {
-    id: 4,
-    leaveType: "Maternity Leave",
-    startDate: "2024-12-01",
-    endDate: "2025-05-31",
-    status: LeavesRequestStatus.Pending,
-    reason: "Childbirth",
-    requestedBy: "Sarah Brown",
-  },
-  {
-    id: 5,
-    leaveType: "Unpaid Leave",
-    startDate: "2024-11-20",
-    endDate: "2024-11-25",
-    status: LeavesRequestStatus.Reject,
-    reason: "Travel",
-    requestedBy: "Michael Lee",
-  },
-];
-
+import { useRouter, useSearchParams } from "next/navigation";
+import { getAllLeaveRequests } from "../actions/get-all-leave-requests";
+import { toast } from "@/hooks/use-toast";
+import { DataSpinner } from "@/common/Loader/Loader";
+import { Routes } from "@/lib/routes";
 
 export const LeavesRequestList = () => {
+  const route = useRouter();
+  const searchParams = useSearchParams();
+  const searchQuery = searchParams.get("key");
+  const [isPending, startTransition] = useTransition();
+  const [filteredLeaves, setFilteredLeaves] = useState<LeaveRequestsTypes[]>(
+    []
+  );
   const columns: ColumnDef<LeaveRequestsTypes>[] = useMemo(
     () => [
       {
         accessorKey: "requestedBy",
         header: "Requested By",
         cell: ({ row }) => (
-          <div className="capitalize">{row.getValue("requestedBy")}</div>
+          <div className="capitalize">
+            {row.getValue("requestedBy") ?? "--"}
+          </div>
         ),
       },
       {
         accessorKey: "leaveType",
         header: "Leave Type",
         cell: ({ row }) => (
-          <div className="capitalize">{row.getValue("leaveType")}</div>
+          <div className="capitalize">{row.getValue("leaveType") ?? "--"}</div>
         ),
       },
       {
         accessorKey: "startDate",
         header: "Start Date",
-        cell: ({ row }) => <div>{row.getValue("startDate")}</div>,
+        cell: ({ row }) => <div>{row.getValue("startDate") ?? "--"}</div>,
       },
       {
         accessorKey: "endDate",
         header: "End Date",
-        cell: ({ row }) => <div>{row.getValue("endDate")}</div>,
+        cell: ({ row }) => <div>{row.getValue("endDate") ?? "--"}</div>,
       },
       {
         accessorKey: "status",
         header: "Status",
         cell: ({ row }) => (
-          <LeavesStatusBadge status={row.getValue("status")} />
+          <LeavesStatusBadge status={row.getValue("status") ?? "--"} />
```

Comment:
why all cell can be undefined ??

Reply:
Actually I have used `nullish` because in database these some missing values But Now I have fixed this